### ISSUE TITLE
Implement support of binding

### ISSUE THREAD

**Issue by mrfranta** [mrfranta/jop#19]
Because a lot of implementations will be "binded" to some annotation, it have to be implement some binding logic where the Annotation can be binded to some class (implementation of some interface). For inspiration look at binding implementations to their interfaces in Google Guice (IOC).

**Comment by mrfranta** [comment]
Nearly finished, last step is implementation of abstract binding factory.

**Comment by mrfranta** [comment]
Posponed. The binding factory will be designed after #33

**Comment by mrfranta** [comment]
Interface for binding factory is done, now can be created abstract binding factory which uses Binder, Bindings...

**Comment by mrfranta** [comment]
Moved implementation for binding factories into another issue. Resolved.
